client/landing/gutenboarding/gutenboard.tsx

```diff
@@ -66,9 +66,16 @@ export function Gutenboard() {
 		: undefined;
 	const goToPrevStep = currentStep > 0 ? () => setStep( step => step - 1 ) : undefined;
 
-	const onboardingBlock = useMemo( () => createBlock( name, { step: currentStep } ), [
-		currentStep,
-	] );
+	const onboardingBlock = useRef( createBlock( name, { step: currentStep } ) );
+
+	const { updateBlockAttributes } = useDispatch( 'core/block-editor' );
+	useEffect(
+		() =>
+			void updateBlockAttributes( onboardingBlock.current.clientId, {
+				step: currentStep,
+			} ),
+		[ currentStep, updateBlockAttributes ]
```

Comment:
Do we expect `updateBlockAttributes` to change? :thinking:

Reply:
It think it will be stable, but the exhaustive deps linter wants you to include it and the cost of including it is probably negligible.